### PR TITLE
F/cloud 1673 dynamodb

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Dynamo db module

Fixes [#2](https://usxtech.atlassian.net/browse/CLOUD-1673)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create terraform.tfvars.json (provided in examples)
  - Clone/Pull branch
Test Case 1 [Create Dynamo db table]:
- Steps
  - Terraform init
  - Terraform plan/apply
- Result
 - Should create table in respective account

Test Case 2 [Destroy Dynamo db table]:
- Steps
  - terraform destroy
- Results
  - Should delete the bucket

```

- [ ] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

**Comment by rchengvariant** [comment]
This looks good. Personally, I would remove the variables with default from examples/terraform.tfvars.json since we want a minimal input. Developers probably don't want to put `PAY_PER_REQUEST` and such in their tfvars.

File: examples/terraform.tfvars.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+    "table_name": "test-11",
+    "billing_mode": "PAY_PER_REQUEST",
+    "hash_key": "UserId",
+    "range_key": "Name",
```
[the remark this answers is not in the document]
Comment (naveen-variant) [reply examples/terraform.tfvars.json]: Removed and added with simple and complex examples.

File: examples/vars/terraform-example-1.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+    "table_name": "test-11",
```
Comment (rchengvariant) [inline examples/vars/terraform-example-1.tfvars.json:2]: Can you rename these files to something else? Maybe `*-detailed` and `-simplified`

**Comment by rchengvariant** [comment]
```
Pre reqs:
  - Create terraform.tfvars.json (provided in examples)
  - Clone/Pull branch
Test Case 1 [Create Dynamo db table]:
- Steps
  - Terraform init
  - Terraform plan/apply
- Result
 - Should create table in respective account [PASSED]

Test Case 2 [Destroy Dynamo db table]:
- Steps
  - terraform destroy
- Results
  - Should delete the table [PASSED]
  ```

**Review by rchengvariant — APPROVED** [review]

**Pull request merged**